axon-server-connector/src/main/java/org/axonframework/axonserver/connector/ErrorCode.java

```diff
@@ -16,84 +16,99 @@
 
     // Internal errors
-    DATAFILE_READ_ERROR( "AXONIQ-9000", EventStoreException.class),
-    INDEX_READ_ERROR( "AXONIQ-9001", EventStoreException.class),
-    DATAFILE_WRITE_ERROR( "AXONIQ-9100", EventStoreException.class),
-    INDEX_WRITE_ERROR( "AXONIQ-9101", EventStoreException.class),
-    DIRECTORY_CREATION_FAILED("AXONIQ-9102", EventStoreException.class),
-    VALIDATION_FAILED( "AXONIQ-9200", EventStoreException.class),
-    TRANSACTION_ROLLED_BACK( "AXONIQ-9900", AggregateRolledBackException.class),
-    OTHER( "AXONIQ-0001", EventStoreException.class),
-    ;
-
-    private final String code;
-    private final Class exceptionClass;
-
-    ErrorCode(String code, Class<? extends AxonException> exceptionClass) {
-        this.code = code;
-        this.exceptionClass = exceptionClass;
+    DATAFILE_READ_ERROR( "AXONIQ-9000", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+    INDEX_READ_ERROR( "AXONIQ-9001", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+    DATAFILE_WRITE_ERROR( "AXONIQ-9100", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+    INDEX_WRITE_ERROR( "AXONIQ-9101", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+    DIRECTORY_CREATION_FAILED("AXONIQ-9102", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+    VALIDATION_FAILED( "AXONIQ-9200", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+    TRANSACTION_ROLLED_BACK( "AXONIQ-9900", (code, error) -> new EventStoreException(error.getMessage(), new AxonServerException(code, error))),
+
+    //Default
+    OTHER("AXONIQ-0001", AxonServerException::new);
+
+    private final String errorCode;
+    private final BiFunction<String, ErrorMessage, ? extends AxonException> exceptionFn;
+
+    ErrorCode(String errorCode, BiFunction<String, ErrorMessage, ? extends AxonException> exceptionFn) {
+        this.errorCode = errorCode;
+        this.exceptionFn = exceptionFn;
     }
 
-
-    public static Class<? extends AxonException> lookupExceptionClass(String code) {
-        return Arrays.stream(values()).filter(mapping -> mapping.code.equals(code))
-                .map(mapping -> mapping.exceptionClass)
-                .findFirst().orElse(OTHER.exceptionClass);
+    private static AxonException convert(String code, Throwable t) {
+        return stream(values()).filter(mapping -> mapping.errorCode.equals(code))
+              .findFirst()
+              .orElse(OTHER)
+              .convert(ExceptionSerializer.serialize("", t));
     }
 
     public static AxonException convert(Throwable t) {
         if( t instanceof EventStoreClientException) {
```

Comment:
Shouldn't we try to convert exceptions other than `EventStoreClientException` in this way? Also, a comment on this method might be helpful.